### PR TITLE
Popup window

### PR DESCRIPTION
**Pull request by shafiuyushawu**
### **Changes made ♻️**

- Created a popup page using JS DOM
- Linked the popup page to my project card using `e.addEventListners()`
- Added `html`, `css` and `js` linters

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -147,242 +148,134 @@ <h3>Multi-Post Stories</h3>
...
+                    <h3>Professional Art <br> Printing Data</h3>
+                    <p>
+                      A daily selection of privatelys <br> personalized reads; no accounts or <br> sign-ups required. has been the <br> industry's standard
+                    </p>
+                    <div class="profession__project-butons">
+                      <button type="button">HTML</button>
+                      <button type="button">Boostrap</button>
+                      <button type="button">Ruby</button>
+                    </div>
+                  </div>
```
Comment (Qoosim) [inline index.html]: - [x] It is required that the projects section is created dynamically with the information from the projects array (all the HTML of the section is created when the page loads). This requires changes in existing code - already created HTML code for projects list should be removed, and instead it should be added via JavaScript. Kindly remove all the already html created codes and create each project with its information dynamically from javascript array of objects from your js file.

project-rqt

File: popup.css [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+  gap: 20px;
+  position: relative;
+  margin: 24px;
+  width: 95%;
+  height: 90vh;
+  background-color: white;
+  flex-direction: column;
+  padding: 16px;
+}
+
```
Comment (Qoosim) [inline popup.css]: - [x] It is required to stick to the guiding design as close as possible. In your popup window, there's more than required space between project image and the description. Kindly fix that.
- [x] The width of popup box is wider than what we have on the Figma template. Kindly reduce the width to match the figma design.
- [x] Also, there's more than required left-margin at the left side of the project image. Kindly look into this and fix.

### Your Design
your-popup-desktop

### Figma Design
temp-popup

 

- [ ] For the mobile version, the whole information on the popup card is not visible and while trying to scroll up down to see the rest of the information, there is overlay information and the rest of the information cannot be seen. Please, take a look.
[your-popup-mobile.webm](https://user-images.githubusercontent.com/26597432/216333821-fcf6be45-a870-4d8d-8503-9b4e3f4d03f8.webm)

**Review by Qoosim — CHANGES_REQUESTED** [review]
#  *Status: Required Changes* :recycle: 

Hi @shafiuyushawu, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
-  No linter errors. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Correct GitHub Flow. :heavy_check_mark: 



## Required Changes ♻️

- [x] Kindly follow this [link](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/adding-a-license-to-a-repository) to set up license for your project. At this moment, your MIT license is not properly set it. Below is the sample how the page that would open when the license link is clicked would look like.
### Sample
sample-license

### Yours 
license


_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by huseyinbiyik — CHANGES_REQUESTED** [review]
Hi team

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.
tenor

 ## Highlights
- Linter set up well and checks pass 🟢
- Great teamwork 👌👌



 ## Required Changes ♻️
- You did a great job but unfortunately your See Project button is not working. So I couldn't check the requested design from the previous reviewer. But for sure, the next one will check them.
Screen Recording 2023-02-02 at 8 45 58 PM

- As the previous reviewer requested, please remove projects section from your HTML file. Popup and Projects section needs to be created dynamically.

- Great job fixing the License file. But let's make sure the license link on your README directs you to the license file. Currently, I am directed to the 404 page.∂



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: popup.js [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+
+for (let popup = 0; popup < openPopups.length; popup += 1) {
+  openPopups[popup].addEventListener('click', () => {
+    const mainDiv = document.createElement('div');
+    mainDiv.className = 'section';
+    const popup = document.createElement('div');
+    popup.className = 'popup__section';
+
+    const topContainer = document.createElement('div');
+    topContainer.className = 'topContainer';
```
Comment (brainconnect93) [inline popup.js]: - [x] As suggested by the previous reviewers, i would kindly suggest you stick to the project requirement as well as the Figma design as much as possible, Currently your project is not functioning the way it should function.

  - Only 2 out of the 7 `see project` buttons are functioning i.e display the popup card when clicked on, while the remaining 5 are not functioning at all.
  - Kindly make sure your popup cards align well to the screen and be sure it fit to the center of the screen as suggested by the previous reviewer and stated in the Figma design.
  - I would kindly suggest you work on the `mobile view` of the popup cards as it is not aligning properly to the mobile screen size of `375px`

Below is the video illustration of how your project looks like 👇 

[screen-capture (5).webm](https://user-images.githubusercontent.com/78634841/216482208-891687b0-2c74-4210-ab41-757fa9058c75.webm)

**Review by brainconnect93 — CHANGES_REQUESTED** [review]
# STATUS: Changes Required :recycle:

### Hi @shafiuyushawu, 👋 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



### To Highlight :1st_place_medal: 
-  Has a proper Pull Request. ✔️  
- No Linter errors :heavy_check_mark: 
- Good use of Github flow ✔️ 

### Changes Required

- [x]  _Please check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me on( @brainconnect93 ) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: popup.js [hunk 4]
```diff
@@ -0,0 +1,366 @@
...
+    seeSourceLink.appendChild(seeSource);
+    seeSource.appendChild(seeSourceIcon);
+    mainDiv.appendChild(popup);
+    body.appendChild(mainDiv);
+
+    close.addEventListener('click', () => {
+      mainDiv.remove();
+    });
+  });
+}
```
Comment (Ibizugbe) [inline popup.js]: I can see you have made lots of efforts in making the changes requested by the previous revious revieweers👍👍 But I can still see some issues with the popup menu that hasn't been resolved.

- On deskop view, your popup still looks different from the given [design](https://www.figma.com/file/l7SqJ3ZfkAKih9sFxvWSR4/Microverse-Student-Project-1?node-id=1%3A1471&t=o7LCculjg0UhWvGh-0)😉😉 currently, yoour image is flowing out of the popup container and from the design, the buttons are not in the right position, kindly compare the design or the [images](https://user-images.githubusercontent.com/26597432/216331796-9e0a12a2-5e86-43d4-87a1-cadb1a677356.png) displayed by the previous reviewers to yours.
Your popup(desktop)
Screenshot from 2023-02-03 15-40-20

- On mobile,your popup menu still looks different from the design.
your popup(mobile)
Screenshot from 2023-02-03 15-39-55

design
image


don't forget that it is a requirement in [this milestone](https://github.com/microverseinc/curriculum-html-css/blob/main/portfolio/8_popup.md) that we stick to the [design](https://www.figma.com/file/l7SqJ3ZfkAKih9sFxvWSR4/Microverse-Student-Project-1?node-id=1%3A1471&t=o7LCculjg0UhWvGh-0) even as we make our page responsive😉😉. kindly make the necessary adjustments🤗🤗

**Review by Ibizugbe — CHANGES_REQUESTED** [review]
Hi @shafiuyushawu 👋, 

68747470733a2f2f6d65646961322e67697068792e636f6d2f6d656469612f4a713832345239334a734c775a436169534c2f67697068792e676966


Good job so far!👍
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights✅✅✅

- [ ] Linters are passing👍.
- [ ] GIthub flow practice has been used👍.


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- [OPTIONAL] Currently your page is overflowing on mobile screen, I suggest you take a look at it as this can lead to a frustrating user experience if not properly maintained👍👍 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by huseyinbiyik — APPROVED** [review]
Hi @shafiuyushawu

## ✔️ Status: APPROVED 🏆 🎉 🎊

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Alt Text



## Highlights
- Great implementation of the requested changes of previous reviewers 🕸️
- Project and popup content created dynamically 👌

## Optional suggestions
- Great job but let's fix the horizontal overflow on mobile version. So our web page will more responsive 👌 Please check the GIF below to see the issue. Also please make sure that popup is aligned to the middle of the screen properly 📱
Screen Recording 2023-02-03 at 10 03 23 PM


_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**